fe/src/main/java/org/apache/doris/clone/DynamicPartitionScheduler.java

```diff
@@ -168,98 +281,62 @@ private void dynamicAddPartition() {
-                        PartitionKey upperBound = PartitionKey.createPartitionKey(Collections.singletonList(upperValue), Collections.singletonList(partitionColumn));
-                        addPartitionKeyRange = Range.closedOpen(lowerBound, upperBound);
-                    } catch (AnalysisException e) {
-                        // keys.size is always equal to column.size, cannot reach this exception
-                        LOG.error("Keys size is not equl to column size.");
-                        continue;
-                    }
-                    for (Range<PartitionKey> partitionKeyRange : info.getIdToRange().values()) {
-                        // only support single column partition now
-                        try {
-                            RangeUtils.checkRangeIntersect(partitionKeyRange, addPartitionKeyRange);
-                        } catch (DdlException e) {
-                            isPartitionExists = true;
-                            if (addPartitionKeyRange.equals(partitionKeyRange)) {
-                                clearFailedMsg(olapTable.getName());
-                            } else {
-                                recordFailedMsg(olapTable.getName(), e.getMessage());
-                            }
-                            break;
-                        }
-                    }
-                    if (isPartitionExists) {
-                        continue;
-                    }
-
-                    // construct partition desc
-                    PartitionKeyDesc partitionKeyDesc = new PartitionKeyDesc(Collections.singletonList(lowerValue), Collections.singletonList(upperValue));
-                    HashMap<String, String> partitionProperties = new HashMap<>(1);
-                    partitionProperties.put("replication_num", String.valueOf(DynamicPartitionUtil.estimateReplicateNum(olapTable)));
-                    SingleRangePartitionDesc rangePartitionDesc = new SingleRangePartitionDesc(true, partitionName,
-                            partitionKeyDesc, partitionProperties);
-
-                    // construct distribution desc
-                    HashDistributionInfo hashDistributionInfo = (HashDistributionInfo) olapTable.getDefaultDistributionInfo();
-                    List<String> distColumnNames = new ArrayList<>();
-                    for (Column distributionColumn : hashDistributionInfo.getDistributionColumns()) {
-                        distColumnNames.add(distributionColumn.getName());
-                    }
-                    DistributionDesc distributionDesc = new HashDistributionDesc(dynamicPartitionProperty.getBuckets(), distColumnNames);
-
-                    // add partition according to partition desc and distribution desc
-                    addPartitionClauses.add(new AddPartitionClause(rangePartitionDesc, distributionDesc, null, false));
-                }
+                addPartitionClauses = getAddPartitionClause(olapTable, partitionColumn, partitionFormat);
```

Comment:
No need to get add partition clause is `skipAddPartition` is true.

Reply:
done